sensorserviceutils/sensorserviceutils.go

```diff
@@ -567,6 +674,30 @@ func (p *parser) extractRegistrationHistory() ([]error, []error) {
 					// For active connection, set current time as the end time
 					// for the ongoing subscription event.
 					sensorName := p.sensors[sensorNumber].Name
+					if samplingRateHz != conn.SamplingRateHz {
```

Comment:
This function is too long and too far indented with various if conditions. Split this function for readability